modules/userdirectory/src/main/java/org/opencastproject/userdirectory/JpaGroupRoleProvider.java

```diff
@@ -280,18 +279,9 @@ public void updateGroupMembershipFromRoles(String userName, String orgId, List<S
         //ignore groups of other providers
         continue;
       }
-      try {
-        if (roleList.contains(group.getRole())) {
-          // record this membership
-          membershipRoles.add(group.getRole());
-        } else {
-          // remove user from this group
-          logger.debug("Removing user {} from group {}", userName, group.getRole());
-          group.getMembers().remove(userName);
-          addGroup(group);
-        }
-      } catch (UnauthorizedException e) {
-        logger.warn("Unauthorized to add or remove user {} from group {}", userName, group.getRole(), e);
+      if (roleList.contains(group.getRole())) {
```

Comment:
This *seems* reasonable, but breaks things in at least one subtle way:

- `JpaUserAndRoleProvider`:496 uses `updateGroupMembershipFromRoles` to set the group membership to an empty set before deleting the user.  Removing the removal here will leave dangling references.